onebox.sh

```diff
@@ -0,0 +1,296 @@
+readonly MSFT="microsoft"
+
+##########################
+# Script Defaults that can be overriden via
+# - parameter arguments OR
+# - assignment here
+##########################
+TEMPLATE_TYPE=fullstack # or devstack
+BRANCH_VERSIONS=edge    # or stable or edx
+DEFAULT_PASSWORD=
+
+MONGO_USER=oxamongoadmin
+MONGO_PASSWORD=$DEFAULT_STRING
+
+MYSQL_ADMIN_USER=root
+MYSQL_ADMIN_PASSWORD=
+
+MYSQL_USER=oxamysql
+MYSQL_PASSWORD=$DEFAULT_STRING
+
+EDXAPP_SU_USERNAME=edx_admin
+EDXAPP_SU_PASSWORD=$DEFAULT_STRING
+
+##########################
+# Settings
+##########################
+readonly BASE_URL=$HOSTNAME
+readonly LMS_URL=$BASE_URL # vanity
+readonly CMS_URL=$BASE_URL
+readonly PREVIEW_URL=$BASE_URL
+readonly PLATFORM_NAME="$MSFT Learning on $HOSTNAME"
+readonly EDXAPP_IMPORT_KITCHENSINK_COURSE=true
+readonly EDXAPP_ENABLE_THIRD_PARTY_AUTH=false
+readonly EDXAPP_SU_EMAIL="${EDXAPP_SU_USERNAME}@${MSFT}.com"
+readonly PLATFORM_EMAIL="$EDXAPP_SU_EMAIL"
+readonly EDX_BRANCH="open-release/ficus.master"
+
+##########################
+# Script Parameter Arguments
+##########################
+parse_args() 
+{
+    while [[ "$#" -gt 0 ]] ; do
+        arg_value="${2}"
+        shift_once=0
+
+        if [[ "${arg_value}" =~ "--" ]] ; then
+            arg_value=""
+            shift_once=1
+        fi
+
+         # Log input parameters to facilitate troubleshooting
+        echo "Option '${1}' set with value '"${arg_value}"'"
+
+        case "$1" in
+          -r|--role)
+            TEMPLATE_TYPE="${arg_value,,}" # convert to lowercase
+            ;;
+          -b|--branches)
+            BRANCH_VERSIONS="${arg_value,,}" # convert to lowercase
+            ;;
+          -d|--default-password)
+            DEFAULT_PASSWORD="${arg_value}"
+            ;;
+
+          --mongo-user)
+            MONGO_USER="${arg_value}"
+            ;;
+          --mongo-password)
+            MONGO_PASSWORD="${arg_value}"
+            ;;
+          --mysql-admin-user)
+            MYSQL_ADMIN_USER="${arg_value}"
+            ;;
+          --mysql-admin-password)
+            MYSQL_ADMIN_PASSWORD="${arg_value}"
+            ;;
+          --mysql-user)
+            MYSQL_USER="${arg_value}"
+            ;;
+          --mysql-password)
+            MYSQL_PASSWORD="${arg_value}"
+            ;;
+          --edxapp-su-username)
+            EDXAPP_SU_USERNAME="${arg_value}"
+            ;;
+          --edxapp-su-password)
+            EDXAPP_SU_PASSWORD="${arg_value}"
+            ;;
+          *)
+            # Unknown option encountered
+            echo "Option '${BOLD}$1${NORM} ${arg_value}' not allowed."
+            ;;
+        esac
+
+        shift # past argument or value
+
+        if [[ $shift_once -eq 0 ]] ; then
+            shift # past argument or value
+        fi
+
+    done
+}
+
+fix_args()
+{
+    # Harden credentials if none were provided.
+    set +x
+    MONGO_PASSWORD=`harden $MONGO_PASSWORD`
+    #MYSQL_ADMIN_PASSWORD=`harden $MYSQL_ADMIN_PASSWORD`
```

Comment:
FYI: i'll open a bug that re-enables this. there are some minor edge cases the need to be addressed. i'll reply to this comment with the bug number later today